src/release.ts

```diff
@@ -0,0 +1,78 @@
+import chalk from 'chalk';
+import { EOL } from 'os';
+import { runAsPromise } from './utils/process';
+
+const yargs = require('yargs');
+
+const {
+	'release': releaseVersion,
+	'next': nextVersion,
+	'dry-run': dryRun,
+	'tag': releaseTag
+} = yargs
+	.option('release', {
+		type: 'string'
+	})
+	.option('next', {
+		type: 'string'
+	})
+	.option('dry-run', {
+		type: 'boolean',
+		default: false
+	})
+	.option('tag', {
+		type: 'string',
+		default: 'next'
+	})
+	.argv;
+
+async function command(bin: string, args: string[], options: any = {}, executeOnDryRun = false) {
+	if (dryRun && !executeOnDryRun) {
+		if (options.cwd) {
+			console.log(chalk.gray(`(from ${options.cwd}) ${bin} ${args.join(' ')}`));
+		}
+		else {
+			console.log(chalk.gray(`${bin} ${args.join(' ')}`));
+		}
+		return Promise.resolve('');
+	}
+
+	console.log(chalk.green(`${bin} ${args.join(' ')}...`));
+
+	return runAsPromise(bin, args, options);
+}
+
+(async function () {
+	const hasDojoRemote = (await command('git', ['remote'], {}, true))
+		.split(EOL)
+		.filter((remote: string) => remote.trim() === 'dojo').length === 1;
+
+	console.log(chalk.yellow(`Version: ${releaseVersion}`));
+	console.log(chalk.yellow(`Next Version: ${nextVersion}`));
+	console.log(chalk.yellow(`Dry Run: ${dryRun}`));
+	console.log(chalk.yellow(`Push Back: ${hasDojoRemote}`));
+
+	// update the version
+	await command('npm', ['version', releaseVersion, '--no-git-tag-version'], { cwd: 'dist/release' }, false);
+	await command('npm', ['version', releaseVersion, '--no-git-tag-version'], {}, false);
```

Comment:
If we remove `--no-git-tag-version` (apart from dry runs) then it should do the git tag for us (and then we won't need the `git tag` command below.

Reply:
Yeah, so I did this on purpose. I didn't want the tag to be created if the `npm publish` task didn't work.